typescript/core/src/McapStreamReader.ts

```diff
@@ -50,7 +50,9 @@ type McapReaderOptions = {
  * ```
  */
 export default class McapStreamReader {
-  #buffer = new StreamBuffer(MCAP_MAGIC.length * 2);
+  #buffer = new ArrayBuffer(MCAP_MAGIC.length * 2);
+  #view = new DataView(this.#buffer);
+  #reader = new Reader(this.#view, MCAP_MAGIC.length * 2); // Cursor starts at end of initial buffer
```

Comment:
I guess I didn't catch this in the original PR review, but why do we start with an offset that is not 0? Perhaps we should start with `#view = new DataView(this.#buffer, 0, 0)` instead and then start the reader with an offset of 0? (Not sure if that will work but it makes more sense to me)

Reply:
Agreed, it looks odd. I will change it to `#view = new DataView(this.#buffer, 0, 0)`, even though I think that starting with an empty buffer would be also OK